src/main/java/com/artipie/asto/FileStorage.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto;
+
+import com.jcabi.log.Logger;
+import hu.akarnokd.rxjava3.jdk8interop.CompletableInterop;
+import hu.akarnokd.rxjava3.jdk8interop.SingleInterop;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Single;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Flow;
+import java.util.stream.Collectors;
+import org.reactivestreams.FlowAdapters;
+
+/**
+ * Simple storage, in files.
+ *
+ * @since 0.1
+ */
+public final class FileStorage implements Storage {
+
+    /**
+     * Where we keep the data.
+     */
+    private final Path dir;
+
+    /**
+     * Ctor.
+     *
+     * @param path The path to the dir
+     */
+    public FileStorage(final Path path) {
+        this.dir = path;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> exists(final String key) {
+        return (CompletableFuture<Boolean>) Single.fromCallable(
```

Comment:
@Sammers21 I don't think we need to cast `CompletionStage` to `CompletableFuture`, since stage interface has `CompletableFuture<T> toCompletableFuture()` method which returns self in case if the stage is a future or transforms itself into future if it is not. See https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html#toCompletableFuture--

So it would be OK to return `CompletionStage` here.